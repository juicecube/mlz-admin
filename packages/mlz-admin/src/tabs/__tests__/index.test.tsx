import React, { Component } from 'react';
import { testMount, testSnapshot } from '../../../../../tests';
import Tabs from '..';
import Menu from '../../menu';
import { mount } from 'enzyme';

const paneList = [
  {
    tab: 'Tab 1',
    key: '1',
    children: 'Content of Tab Pane 1',
  },
  {
    tab: 'Tab 2',
    key: '2',
    children: 'Content of Tab Pane 2',
  },
  {
    tab: 'Tab 3',
    key: '3',
    children: 'Content of Tab Pane 3',
  },
  {
    tab: 'Tab 4',
    key: '4',
    children: 'Content of Tab Pane 4',
  },
];

const menuList = [
  {
    title: 'tab1',
  },
  {
    title: 'tab2',
  },
  {
    title: 'tab3',
  },
];

const { TabPane, ContextMenuTabs } = Tabs;
const TempTabsMounter = () => {
  return (
    <Tabs defaultActiveKey="1">
      <TabPane tab="Tab 1" key="1">
        Content of Tab Pane 1
      </TabPane>
      <TabPane tab="Tab 2" key="2">
        Content of Tab Pane 2
      </TabPane>
      <TabPane tab="Tab 3" key="3">
        Content of Tab Pane 3
      </TabPane>
    </Tabs>
  );
};

const contextMenuTabsWrapper = mount(
  <ContextMenuTabs
    hideAdd
    contextMenu={
      <Menu>
        {menuList.map((item) => {
          return <Menu.Item key={item.title}>{item.title}</Menu.Item>;
        })}
      </Menu>
    }>
    {paneList.map((item) => {
      return (
        <TabPane tab={item.tab} key={item.key}>
          {item.children}
        </TabPane>
      );
    })}
  </ContextMenuTabs>,
);

describe('🧪 Tabs', () => {
  //
  testMount(TempTabsMounter);
  testSnapshot(TempTabsMounter);

  // contextMenu属性设置内容时，渲染对应的tabbar
  test('ContextMenuTabs正确浅渲染子元素', () => {
    expect(contextMenuTabsWrapper.find('.admini-tabs__tab-pane').length).toBe(paneList.length);
  });

  // contextMenu属性设置内容时，且唤起contextMenu时展示菜单
  test('ContextMenuTabs的contextMenu属性正确被触发', () => {
    contextMenuTabsWrapper
      .find('.admini-tabs__tab-pane')
      .first()
      .simulate('contextmenu');
    expect(document.getElementsByClassName('ant-dropdown-menu-item').length).toBe(menuList.length);
  });

  // contextMenu属性设置内容时，点击tab标签页切换指定标签页且正确触发事件
  test('ContextMenuTabs的Tab点击后可以正确切换', () => {
    contextMenuTabsWrapper
      .find('.admini-tabs__tab-pane')
      .at(1)
      .simulate('click');
    expect(contextMenuTabsWrapper.find('.admini-tabs__tab-pane').length).toBe(paneList.length);
    expect(
      contextMenuTabsWrapper
        .find('.admini-tabs__tab-pane')
        .at(1)
        .hasClass('active'),
    ).toBeTruthy();
  });

  // contextMenu属性设置内容时，点击add标签页正确触发事件
  test('ContextMenuTabs的Tab点击后正确触发事件', () => {
    const onTabClickHandler = jest.fn();
    const onChangeHandler = jest.fn();

    const tempContextMenuTabsWrapper = mount(
      <ContextMenuTabs
        hideAdd
        onTabClick={onTabClickHandler}
        onChange={onChangeHandler}
        contextMenu={
          <Menu>
            {menuList.map((item) => {
              return <Menu.Item key={item.title}>{item.title}</Menu.Item>;
            })}
          </Menu>
        }>
        {paneList.map((item) => {
          return (
            <TabPane tab={item.tab} key={item.key}>
              {item.children}
            </TabPane>
          );
        })}
      </ContextMenuTabs>,
    );

    const indexNumber = 1;
    tempContextMenuTabsWrapper
      .find('.admini-tabs__tab-pane')
      .at(indexNumber)
      .simulate('click');
    expect(onTabClickHandler).toHaveBeenCalled();
    // 一次是初始化时的变化，一次是click时的变化
    expect(onChangeHandler).toHaveBeenCalledTimes(2);
  });

  // TODO: contextMenu属性设置内容时，点击关闭和新增按钮触发对应的onEdit事件
});
