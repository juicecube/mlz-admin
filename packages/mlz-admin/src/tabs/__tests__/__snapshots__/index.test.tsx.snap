// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🧪 Tabs snapshot test 快照匹配 1`] = `"<div class=\\"ant-tabs ant-tabs-top\\"><div role=\\"tablist\\" class=\\"ant-tabs-nav\\"><div class=\\"ant-tabs-nav-wrap\\"><div class=\\"ant-tabs-nav-list\\" style=\\"transform: translate(0px, 0px);\\"><div class=\\"ant-tabs-tab ant-tabs-tab-active\\"><div role=\\"tab\\" aria-selected=\\"true\\" class=\\"ant-tabs-tab-btn\\" tabindex=\\"0\\" id=\\"rc-tabs-test-tab-1\\" aria-controls=\\"rc-tabs-test-panel-1\\">Tab 1</div></div><div class=\\"ant-tabs-tab\\"><div role=\\"tab\\" aria-selected=\\"false\\" class=\\"ant-tabs-tab-btn\\" tabindex=\\"0\\" id=\\"rc-tabs-test-tab-2\\" aria-controls=\\"rc-tabs-test-panel-2\\">Tab 2</div></div><div class=\\"ant-tabs-tab\\"><div role=\\"tab\\" aria-selected=\\"false\\" class=\\"ant-tabs-tab-btn\\" tabindex=\\"0\\" id=\\"rc-tabs-test-tab-3\\" aria-controls=\\"rc-tabs-test-panel-3\\">Tab 3</div></div><div class=\\"ant-tabs-ink-bar ant-tabs-ink-bar-animated\\"></div></div></div><div class=\\"ant-tabs-nav-operations ant-tabs-nav-operations-hidden\\"><button type=\\"button\\" class=\\"ant-tabs-nav-more\\" style=\\"visibility: hidden; order: 1;\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" aria-haspopup=\\"listbox\\" aria-controls=\\"rc-tabs-test-more-popup\\" id=\\"rc-tabs-test-more\\" aria-expanded=\\"false\\"><span role=\\"img\\" aria-label=\\"ellipsis\\" class=\\"anticon anticon-ellipsis\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" data-icon=\\"ellipsis\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><path d=\\"M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z\\"></path></svg></span></button></div></div><div class=\\"ant-tabs-content-holder\\"><div class=\\"ant-tabs-content ant-tabs-content-top\\"><div role=\\"tabpanel\\" tabindex=\\"0\\" aria-hidden=\\"false\\" class=\\"ant-tabs-tabpane ant-tabs-tabpane-active\\" id=\\"rc-tabs-test-panel-1\\" aria-labelledby=\\"rc-tabs-test-tab-1\\">Content of Tab Pane 1</div><div role=\\"tabpanel\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"display: none;\\" class=\\"ant-tabs-tabpane\\" id=\\"rc-tabs-test-panel-2\\" aria-labelledby=\\"rc-tabs-test-tab-2\\"></div><div role=\\"tabpanel\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" style=\\"display: none;\\" class=\\"ant-tabs-tabpane\\" id=\\"rc-tabs-test-panel-3\\" aria-labelledby=\\"rc-tabs-test-tab-3\\"></div></div></div></div>"`;
